lxd/device/nic_ovn.go

```diff
@@ -0,0 +1,387 @@
+package device
+
+import (
+	"fmt"
+	"net"
+	"os"
+
+	"github.com/pkg/errors"
+
+	"github.com/lxc/lxd/lxd/cluster"
+	deviceConfig "github.com/lxc/lxd/lxd/device/config"
+	"github.com/lxc/lxd/lxd/dnsmasq/dhcpalloc"
+	"github.com/lxc/lxd/lxd/instance"
+	"github.com/lxc/lxd/lxd/instance/instancetype"
+	"github.com/lxc/lxd/lxd/network"
+	"github.com/lxc/lxd/lxd/network/openvswitch"
+	"github.com/lxc/lxd/lxd/revert"
+	"github.com/lxc/lxd/lxd/util"
+	"github.com/lxc/lxd/shared"
+	"github.com/lxc/lxd/shared/api"
+	log "github.com/lxc/lxd/shared/log15"
+)
+
+type nicOVN struct {
+	deviceCommon
+
+	network network.Network
+}
+
+// getIntegrationBridgeName returns the OVS integration bridge to use.
+func (d *nicOVN) getIntegrationBridgeName() (string, error) {
+	integrationBridge, err := cluster.ConfigGetString(d.state.Cluster, "network.ovn.integration_bridge")
+	if err != nil {
+		return "", errors.Wrapf(err, "Failed to get OVN integration bridge name")
+	}
+
+	return integrationBridge, nil
+}
+
+// validateConfig checks the supplied config for correctness.
+func (d *nicOVN) validateConfig(instConf instance.ConfigReader) error {
+	if !instanceSupported(instConf.Type(), instancetype.Container, instancetype.VM) {
+		return ErrUnsupportedDevType
+	}
+
+	requiredFields := []string{
+		"network",
+	}
+
+	optionalFields := []string{
+		"name",
+		"hwaddr",
+		"host_name",
+		"ipv4.address",
+		"ipv6.address",
+		"maas.subnet.ipv4",
+		"maas.subnet.ipv6",
```

Comment:
We can safely drop those two for now as MAAS cannot manage an OVN network and so will never be the one doing IP management and DNS on it.

Reply:
Fixed